pkg/agent/controller/networkpolicy/reject_test.go

```diff
@@ -0,0 +1,275 @@
+		antreaProxyEnabled bool
+		srcIsLocal         bool
+		dstIsLocal         bool
+		expectRejectType   RejectType
+	}{
+		{
+			name:               "RejectPodLocal",
+			isServiceTraffic:   false,
+			antreaProxyEnabled: true,
+			srcIsLocal:         true,
+			dstIsLocal:         true,
+			expectRejectType:   RejectPodLocal,
+		},
+		{
+			name:               "RejectPodRemoteToLocal",
+			isServiceTraffic:   false,
+			antreaProxyEnabled: true,
+			srcIsLocal:         false,
+			dstIsLocal:         true,
+			expectRejectType:   RejectPodRemoteToLocal,
+		},
+		{
+			name:               "RejectPodLocalToRemote",
+			isServiceTraffic:   false,
+			antreaProxyEnabled: true,
+			srcIsLocal:         true,
+			dstIsLocal:         false,
+			expectRejectType:   RejectPodLocalToRemote,
+		},
+		{
+			name:               "RejectServiceLocal",
+			isServiceTraffic:   true,
+			antreaProxyEnabled: true,
+			srcIsLocal:         true,
+			dstIsLocal:         true,
+			expectRejectType:   RejectServiceLocal,
+		},
+		{
+			name:               "RejectServiceRemoteToLocal",
+			isServiceTraffic:   true,
+			antreaProxyEnabled: true,
+			srcIsLocal:         false,
+			dstIsLocal:         true,
+			expectRejectType:   RejectServiceRemoteToLocal,
+		},
+		{
+			name:               "RejectServiceLocalToRemote",
+			isServiceTraffic:   true,
+			antreaProxyEnabled: true,
+			srcIsLocal:         true,
+			dstIsLocal:         false,
+			expectRejectType:   RejectServiceLocalToRemote,
+		},
+		{
+			name:               "RejectNoAPServiceLocal",
+			isServiceTraffic:   true,
+			antreaProxyEnabled: false,
+			srcIsLocal:         true,
+			dstIsLocal:         true,
+			expectRejectType:   RejectNoAPServiceLocal,
+		},
+		{
+			name:               "RejectNoAPServiceRemoteToLocal",
+			isServiceTraffic:   true,
+			antreaProxyEnabled: false,
+			srcIsLocal:         false,
+			dstIsLocal:         true,
+			expectRejectType:   RejectNoAPServiceRemoteToLocal,
+		},
+		{
+			name:               "RejectServiceRemoteToExternal",
+			isServiceTraffic:   true,
+			antreaProxyEnabled: true,
+			srcIsLocal:         false,
+			dstIsLocal:         false,
+			expectRejectType:   RejectServiceRemoteToExternal,
+		},
+		{
+			name:               "Unsupported pod2pod remote2remote",
+			isServiceTraffic:   false,
+			antreaProxyEnabled: true,
+			srcIsLocal:         false,
+			dstIsLocal:         false,
+			expectRejectType:   Unsupported,
+		},
+		{
+			name:               "Unsupported noAP remote2remote",
+			isServiceTraffic:   true,
+			antreaProxyEnabled: false,
+			srcIsLocal:         false,
+			dstIsLocal:         false,
+			expectRejectType:   Unsupported,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rejectType := getRejectType(tt.isServiceTraffic, tt.antreaProxyEnabled, tt.srcIsLocal, tt.dstIsLocal)
+			assert.Equal(t, tt.expectRejectType, rejectType)
+		})
+	}
+}
+
+func TestGetRejectOFPorts(t *testing.T) {
+	nonTunTunPort := uint32(0)
```

Comment:
nonTunPort？

Reply:
I changed it to unsetPort because 0 is not just tunnel related unset value.